src/domain/headlessCms/persistor.ts

```diff
@@ -0,0 +1,203 @@
+import { InMemoryCache } from '@apollo/client/cache/inmemory/inMemoryCache';
+import { CachePersistor, LocalStorageWrapper } from 'apollo3-cache-persist';
+import { NormalizedCacheObject } from '@apollo/client';
+
+import AppConfig from '../app/AppConfig';
+
+// Using console for logging for simplicity, consider a logging library
+export enum LogLevel {
+  DEBUG,
+  INFO,
+  WARN,
+  ERROR,
+}
+
+class TimedApolloCachePersistorLogger {
+  private logLevel: LogLevel;
+
+  constructor(logLevel: LogLevel = LogLevel.INFO) {
+    this.logLevel = logLevel;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  debug(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.DEBUG) {
+      // eslint-disable-next-line no-console
+      console.debug(message, ...optionalParams);
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  info(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.INFO) {
+      // eslint-disable-next-line no-console
+      console.info(message, ...optionalParams);
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  warn(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.WARN) {
+      // eslint-disable-next-line no-console
+      console.warn(message, ...optionalParams);
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  error(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.ERROR) {
+      // eslint-disable-next-line no-console
+      console.error(message, ...optionalParams);
+    }
+  }
+}
+
+export interface TimedApolloCachePersistorOptions {
+  /**
+   * The time-to-live for the persisted cache in milliseconds.
+   * Defaults to 10 minutes (600000 milliseconds).
+   */
+  persistedCacheTimeToLive?: number;
```

Comment:
How about `persistedCacheTimeToLiveMs` or `persistedCacheTimeMsToLive`? It'd be clear at a glance for the user that this is milliseconds, which would be good for clarity IMHO.